src/cli/AttachmentExport.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ *  Copyright (C) 2019 KeePassXC Team <team@keepassxc.org>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <cstdio>
+#include <cstdlib>
+
+#include "AttachmentExport.h"
+
+#include "Utils.h"
+#include "core/Group.h"
+
+#include <QCommandLineParser>
+#include <QFile>
+
+AttachmentExport::AttachmentExport(FILE* fout)
```

Comment:
I would prefer some way to also optionally specify the output file name instead of always dumping to stdout.

Reply:
I decided to entirely remove dumping to stdout when using `attachment-export` for the following reasons:
1. Symmetry with `attachment-import` which is only able to import from a file, not from stdin.
2. Binary output might mess up the terminal.
3. I found no good way to output the attachment which could both be checked by the tests using `m_stdout->readAll()` but also actually displayed the content to the user. All solutions I tried so far only allowed one of those things (at least without including some kind of workaround like `m_fout` that was used previously in `AttachmentExport`, but is now removed).

Reply:
I concur with this and it's consistent with how the gui handles attachments... as files.

Reply:
Thinking this through... there might be value in a `--stdout` flag on attachment-export for people who want to pipe the contents of an attachment into another command. When that flag is issued then the export_file positional parameter is ignored/optional.